text/0008-ingest-gem-content.md

```diff
@@ -0,0 +1,332 @@
+- Feature Name: Ingest Gem File Contents
+- Start Date: 2023-01-18
+- RFC PR: (leave this empty)
+- Bundler Issue: (leave this empty)
+
+# Summary
+
+Create a process for the purpose of processing and saving the contents of a gem.
+The process will run each time a gem is uploaded to save information about each file in the gem.
+The saved output will include file size, content type, SHA256 checksum, and relative path within the gem.
+In the interest of simplifying this RFC and the feature work that follows, this RFC focuses on only the ingestion and storage of gem content.
+
+# Motivation
+
+It is our hope that we can provide access to individual file data from gem files across versions in a compact and efficient way that supports future features depending on gem content.
+
+Creating a gem content ingestion and storage system will enable rubygems.org to support the security and useability gems in the Ruby Gems ecosystem.
+The gem content repository will be the authoritative source for the content of gems, accessible as individual files without without requiring downloading and unpacking a gem.
```

Comment:
nit: it isn't actually authoritative, the .gem files remain the source of truth (since that's what will actually get installed)

Reply:
At the risk of nitpicking semantics, I meant "authoritative" because rubygems.org is the authority on the content of any given gem. It may not be the definitive source, since the definition of a gem content is "what is in the gem file" but we certainly can say, with authority, that the content we're representing is coming "directly from the authority on rubygems.". I'm looking for a word that we could agree on: verifiable source? authentic source? dependable source? I think authoritative captures it since we're certifying the content is correct by proof of authority over the gems.